cmd/keycloakb/keycloak_bridge.go

```diff
@@ -836,6 +863,63 @@ func config(logger log.Logger) *viper.Viper {
 	return v
 }
 
+func checkMigrationVersion(logger log.Logger, dbParams dbConfig, migrationVersionDBConn CloudtrustDB, minMigrationVersion string) error {
+	var err error
+	migrationVersionDBConn, err = dbParams.openDatabase()
+	if err != nil {
+		logger.Log("msg", "could not create DB connection in order to check the migration version", "DB", dbParams.Database, "error", err)
+		return err
+	}
+	var flywayVersion string
+	row := migrationVersionDBConn.QueryRow(`SELECT version FROM flyway_schema_history ORDER BY installed_rank DESC LIMIT 1;`)
+	err = row.Scan(&flywayVersion)
+	if err != nil {
+		logger.Log("msg", "could not query the flyway_schema_history", "error", err)
+		return err
+	}
+
+	// compare the two versions of the type x.y (major.minor)
+	minMajorRequired := strings.Split(minMigrationVersion, ".")
+	minMajorFlyway := strings.Split(flywayVersion, ".")
+
+	majorRequired, err := strconv.Atoi(minMajorRequired[0])
+	if err != nil {
+		logger.Log("msg", "could not convert string to int", "error", err)
+		return err
+	}
+	minorRequired, err := strconv.Atoi(minMajorRequired[1])
+	if err != nil {
+		logger.Log("msg", "could not convert string to int", "error", err)
+		return err
+	}
+
+	majorFlyway, err := strconv.Atoi(minMajorFlyway[0])
+	if err != nil {
+		logger.Log("msg", "could not convert string to int", "error", err)
+		return err
+	}
+	minorFlyway, err := strconv.Atoi(minMajorFlyway[1])
+	if err != nil {
+		logger.Log("msg", "could not convert string to int", "error", err)
+		return err
+	}
+
+	// it is required for the last script version of the flyway is to "bigger" than the required version
+	if (majorFlyway < majorRequired) || (majorFlyway >= majorRequired && minorFlyway < minorRequired) {
+		logger.Log("msg", "the DB schema is not up-to-date", "DB", dbParams.Database)
+		return fmt.Errorf("DB schema not up-to-date")
+	}
+
+	err = migrationVersionDBConn.Close()
```

Comment:
This statement should be **defer**ed right after connection (not closed in case of error)

(It is more as a best practice than everything else because if there is an error, application will stop and connection will be closed anyway)

Reply:
Another idea : call this method after db connection is open in main function and simply use the connection you passed as a parameter without need to close it

Reply:
Changed with the first suggestion